test/unittests/stack_test.cpp

```diff
@@ -48,3 +52,95 @@ TEST(stack, drop_and_peek)
     EXPECT_EQ(stack.size(), 0);
     EXPECT_TRUE(stack.empty());
 }
+
+TEST(stack, clear)
+{
+    Stack<char> stack;
+    stack.push('a');
+    stack.push('b');
+
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.size(), 2);
+
+    stack.drop(0);
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.size(), 2);
+
+    stack.clear();
+    EXPECT_TRUE(stack.empty());
+    EXPECT_EQ(stack.size(), 0);
+}
+
+TEST(stack, resize)
+{
+    Stack<char> stack;
+    stack.push('a');
+    stack.push('b');
+
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.size(), 2);
+
+    // grow stack
+    stack.resize(4);
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.peek(), 0);
+    EXPECT_EQ(stack.peek(1), 0);
+    EXPECT_EQ(stack.size(), 4);
```

Comment:
could also check that initial elements are not changed

Reply:
Added.